### PR TITLE
File Syntaxes

### PR DESCRIPTION
**Pull request by ItsTheSky**
This is to keep a trace of my work and gather code feedback as it's going on :D
## Tasks:
- [X] File syntax class
- [X] File syntaxes loading
- [X] Add support for Skript as default available syntax
- [X] Add syntax highlight selection in the settings per file type
- [x] Implement syntax highlight changes in the editor
- [x] Add specific file syntax highlight (e.g. configure `test.yml` as JSON and not YAML)
- [x] Add support for theme with syntax & syntax items in the marketplace

### CODE REVIEW

**Comment by GliczDev** [comment]
I think there should be a general Language file for other things than just syntax highlighting, e.g. identifying language in an addon

**Comment by ItsTheSky** [comment]
I'm not sure what you mean by that ... you want to let addons register their own language?

**Comment by GliczDev** [comment]
There should be a `Language` (or something like that) class, which represents a programming language supported in SkEditor. This class would contain data about name, file extensions and syntax highlighting. This would allow addons to check which language is used in a specific file, e.g. supporting more languages in Analyzer depending on built-in SkEditor's programming language system.

**Comment by ItsTheSky** [comment]
More likely an enum then? And "unregistrered" syntaxes are referenced as Custom?

**Comment by GliczDev** [comment]
This wouldn't allow addons to register other languages. Maybe just a class with a constructor (name, extensions, syntax highlighting) which is registered somewhere in api?

**Comment by ItsTheSky** [comment]
I understand what you are asking for, but isn't it the FileSyntax class? Or do you want to separate FileSyntax's properties and the loaded highlighting?

**Comment by GliczDev** [comment]
More like FileSyntaxConfig class, however it should be properly renamed and should contain syntax highlighting definition (instead of FileSyntax class)

File: SkEditor/Controls/MainMenuControl.axaml.cs [hunk 1]
```diff
@@ -47,7 +47,7 @@ private void AssignCommands()
         MenuItemDuplicate.Command = new RelayCommand(() => CustomCommandsHandler.OnDuplicateCommandExecuted(ApiVault.Get().GetTextEditor().TextArea));
         MenuItemComment.Command = new RelayCommand(() => CustomCommandsHandler.OnCommentCommandExecuted(ApiVault.Get().GetTextEditor().TextArea));
 
-        MenuItemRefreshSyntax.Command = new RelayCommand(SyntaxLoader.RefreshSyntax);
+        MenuItemRefreshSyntax.Command = new RelayCommand(() => SyntaxLoader.RefreshSyntax());
```
Comment (NotroDev) [inline SkEditor/Controls/MainMenuControl.axaml.cs:50]: Does

File: SkEditor/Utilities/Syntax/SyntaxLoader.cs [hunk 2]
```diff
@@ -10,129 +10,209 @@
...
+        var fileSyntax = await FileSyntax.LoadSyntax(folder);
+        FileSyntaxes.RemoveAll(x => x.Config.FullIdName.Equals(fileSyntax.Config.FullIdName));
+        foreach (string extension in fileSyntax.Config.Extensions)
         {
-            SetDefaultSyntax();
+            if (SortedFileSyntaxes.ContainsKey(extension)) SortedFileSyntaxes[extension].Remove(fileSyntax);
+
+            if (ApiVault.Get().GetAppConfig().FileSyntaxes.ContainsKey(fileSyntax.Config.LanguageName))
+            {
+                ApiVault.Get().GetAppConfig().FileSyntaxes.Remove(fileSyntax.Config.LanguageName);
```
Comment (NotroDev) [inline SkEditor/Utilities/Syntax/SyntaxLoader.cs:104]: ContainsKey isn't needed, if it isn't in the dictionary the Remove method will just return false

File: SkEditor/Utilities/Syntax/SyntaxLoader.cs [hunk 3]
```diff
@@ -10,129 +10,209 @@
...
-                TextEditor editor = (TextEditor)tab.Content;
-                editor.SyntaxHighlighting = AvaloniaEdit.Highlighting.Xshd.HighlightingLoader.Load(reader, HighlightingManager.Instance);
-            });
-            streamReader.Close();
-            reader.Close();
+                return;
+            }
+
+            extension = Path.GetExtension(currentOpenedFile.Tag.ToString()).ToLower();
+            if (string.IsNullOrWhiteSpace(extension) || !SortedFileSyntaxes.ContainsKey(extension) || !SortedFileSyntaxes.ContainsKey(extension))
```
Comment (NotroDev) [inline SkEditor/Utilities/Syntax/SyntaxLoader.cs:173]: two same conditions

File: SkEditor/Utilities/Syntax/SyntaxLoader.cs (hunk 2 above)
Comment (ItsTheSky) [reply SkEditor/Utilities/Syntax/SyntaxLoader.cs:102]: Oh gotcha, didn't knew that

**Pull request merged**